File: Python3/프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
# def solution(numbers):
#     numbers = [str(i) for i in numbers]
#     numbers = sorted(numbers, key= lambda item: item * 3, reverse=True)
#     answer = "".join(numbers)
#     return answer

def solution(numbers):
    numbers = list(map(str, numbers))
    numbers.sort(key=lambda x: x*3, reverse=True)
    return str(int(''.join(numbers)))
    

File: C++/프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(string a, string b) {
    return a + b > b + a;
}

string solution(vector<int> numbers) {
    vector<string> strNumbers;
    
    string answer = "";
    for (int n : numbers) {
        strNumbers.push_back(to_string(n));
    }
    
    sort(strNumbers.begin(), strNumbers.end(), compare);
    
    if (strNumbers[0] == "0") {
        return "0";
    }
    
    for (string s : strNumbers) {
        answer += s;
    }
    return answer;
}